### PR TITLE
Remove updated groups from atm proc interface

### PR DESCRIPTION
**Pull request by bartgol**
The asymmetry between fields and groups has puzzled more than one person so far. This PR makes the treatment of field groups more similar to that of fields.

Additionally, fix the treatment of required fields/groups in the AtmosphereProcessGroup class.

### CODE REVIEW

File: components/scream/src/control/atmosphere_driver.cpp
Comment (AaronDonahue) [inline components/scream/src/control/atmosphere_driver.cpp:35]: nitpicky, but I think we either want "take care of" or "accomodate"

File: components/scream/src/control/atmosphere_driver.cpp
Comment (AaronDonahue) [inline components/scream/src/control/atmosphere_driver.cpp:29]: Do we explain in these comments what the distinction is between a field and a group?  If not, perhaps one line would make it easier to understand for any newcomers.

File: components/scream/src/control/tests/ad_tests.cpp
Comment (AaronDonahue) [inline components/scream/src/control/tests/ad_tests.cpp:36]: I'm sure there is a good reason, but why do we remove this test?

File: components/scream/src/control/tests/ad_tests.yaml
Comment (AaronDonahue) [inline components/scream/src/control/tests/ad_tests.yaml:12]: By removing this are we no longer testing if it is possible to pass a vector quantity to initialize a vector field?

File: components/scream/src/share/atm_process/atmosphere_process.hpp
Comment (AaronDonahue) [inline components/scream/src/share/atm_process/atmosphere_process.hpp:114]: "methods"

File: components/scream/src/share/atm_process/atmosphere_process.hpp
Comment (AaronDonahue) [inline components/scream/src/share/atm_process/atmosphere_process.hpp:170]: I like this comment

File: components/scream/src/share/atm_process/atmosphere_process_group.cpp
Comment (AaronDonahue) [inline components/scream/src/share/atm_process/atmosphere_process_group.cpp:165]: Is there a better way to say this? I think the copious use of the word "group" is a bit confusing.

File: components/scream/src/share/atm_process/atmosphere_process_group.cpp
Comment (AaronDonahue) [inline components/scream/src/share/atm_process/atmosphere_process_group.cpp:148]: This assumes sequential splitting right?  Will this be an issue when we develop parallel splitting in the model?

**Review by AaronDonahue — COMMENTED** [review]
I left a few comments, they shouldn't hold up the merge but because the Automerge is triggered I'm reviewing as a "comment" in case we want to respond before it is merged.

File: components/scream/src/control/tests/ad_tests.cpp
Comment (bartgol) [reply components/scream/src/control/tests/ad_tests.cpp:36]: I just wanted to keep the test small, I guess, so I removed a couple of fields. But I can see there's value in checking initialization procedures. I'll bring some stuff back.

File: components/scream/src/control/atmosphere_driver.cpp
Comment (bartgol) [reply components/scream/src/control/atmosphere_driver.cpp:29]: Added some pointers

File: components/scream/src/control/tests/ad_tests.yaml
Comment (bartgol) [reply components/scream/src/control/tests/ad_tests.yaml:12]: Added back testing for that feature.

File: components/scream/src/share/atm_process/atmosphere_process_group.cpp
Comment (bartgol) [reply components/scream/src/share/atm_process/atmosphere_process_group.cpp:148]: Added a check.

File: components/scream/src/share/atm_process/atmosphere_process_group.cpp
Comment (AaronDonahue) [inline components/scream/src/share/atm_process/atmosphere_process_group.cpp:153]: Thank you for adding the check, could we add another check that the ScheduleType is Sequential and throw an error if not?  I'm thinking that in the future we may implement other ScheduleTypes we've never heard of and putting in as many hooks as possible that make sure we don't accidentally break the code without a map to find out where would be great.

Comment (bartgol) [reply components/scream/src/share/atm_process/atmosphere_process_group.cpp:153]: Do you really think there will be a schedule type other than Sequential or Parallel?

Comment (AaronDonahue) [reply components/scream/src/share/atm_process/atmosphere_process_group.cpp:153]: none specifically come to mind.  I guess from one perspective, anything we come up with will just be a combination of those two elemental coupling approaches.  I suppose if we decide to do something exotic it will be important that whatever that is be not a new ScheduleType per-se but some set of operations that are Sequential or Parallel.

Comment (bartgol) [reply components/scream/src/share/atm_process/atmosphere_process_group.cpp:153]: Well, the way I see it, a "mixed" approach will be likely consist of sub-groups, so each individual sub-group will still be sequential or parallel.

**Review by AaronDonahue — APPROVED** [review]

**Pull request merged**
